pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java

```diff
@@ -735,7 +765,13 @@ protected SegmentBuildDescriptor buildSegmentInternal(boolean forCommit) {
     if (_isOffHeap) {
       params.withMemoryUsedBytes(_memoryManager.getTotalAllocatedBytes());
     }
-    SegmentCompletionProtocol.Response commitEndResponse = _protocolHandler.segmentCommitEnd(params);
+    SegmentCompletionProtocol.Response commitEndResponse;
+    if (!_indexLoadingConfig.isEnableSplitCommitEndWithMetadata()) {
```

Comment:
nit I would prefer the condition inverted, given that both if and else portions are one line long.